snowshu/adapters/target_adapters/base_target_adapter.py

```diff
@@ -103,33 +105,55 @@ def create_or_replace_view(self, relation) -> None:
         """
         raise NotImplementedError()
 
-    def load_data_into_relation(self, relation: Relation) -> None:
+    def load_data_into_relation(self, relation: Relation, data: pd.DataFrame) -> None:
+        """Loads data into a to target.
+
+        Args:
+            relation: The relation containing info about dataset to load.
+            data: The data to load into the relation.
+        """
+        if data is None and relation.data.empty:
+            raise ValueError("Both data and relation.data are empty.")
```

Comment:
Why the case when relation have no data is invalid? For sure it doesn't seem very useful, but it still can be valid, am I wrong?

Reply:
You're right, I've fixed it to not raise exception but also added some logs to be more informative on replica creation if happens with empty data.